harness/determined/core/_context.py

```diff
@@ -39,6 +39,7 @@ class Context:
 
     def __init__(
         self,
+        _session: api.Session,
```

Comment:
I don't like that I don't understand how this works. I can't tell how much of that is my fault, how much the PR's. When you're ready to talk about this part, could we chat?